File: PYTHON/Numeric-Functions/decimal_to_octal.py
def decimal_to_octal (num):
	num = int(num)
	res = int(0)
	level = int(1)
	while (num > 0):
		tempI = int(num / 8)
		tempF = float(num / 8.0)
		tempF = float(tempF - tempI)
		tempF = int(tempF * 8)
		res = int(res + tempF * level)
		level = int(level  * 10)
		num = int(tempI)
	return res

x = int(input("User, enter a number: "))
results = decimal_to_octal(x)
print(str(x) + " --> " + str(results))

File: JAVA/decimal_to_octal.java
import java.util.Scanner;
class decimal_to_binary {
	public static void main (String [] args) {
		Scanner SC = new Scanner(System.in);
		int x;
		int results;
		System.out.println("User, enter a number: ");
		x = SC.nextInt();
		results = decimal_to_octal(x);
		System.out.println(x + " --> " + results);
	}
	public static int decimal_to_octal (int num) {
		int res = 0;
		int level = 1;
		while (num > 0) {
			int tempI = (int)(num / 8);
			float tempF = (float)(num / 8.0);
			tempF = tempF - tempI;
			tempF = (int)(tempF * 8);
			res += tempF * level;
			level *= 10;
			num = tempI;
		}
	return res;
	}
}
